examples/autoignition-mpi.py

```diff
@@ -0,0 +1,229 @@
+from grudge.shortcuts import make_visualizer
+
+
+from mirgecom.euler import inviscid_operator
+from mirgecom.simutil import (
+    inviscid_sim_timestep,
+    sim_checkpoint,
+    create_parallel_grid,
+    ExactSolutionMismatch
+)
+from mirgecom.io import make_init_message
+from mirgecom.mpi import mpi_entry_point
+
+from mirgecom.integrators import rk4_step
+from mirgecom.steppers import advance_state
+from mirgecom.boundary import AdiabaticSlipBoundary
+from mirgecom.initializers import MixtureInitializer
+from mirgecom.eos import PyrometheusMixture
+from mirgecom.euler import split_conserved
+import cantera
+import pyrometheus as pyro
+
+logger = logging.getLogger(__name__)
+
+
+@mpi_entry_point
+def main(ctx_factory=cl.create_some_context):
+    """Drive example."""
+    cl_ctx = ctx_factory()
+    queue = cl.CommandQueue(cl_ctx)
+    actx = PyOpenCLArrayContext(queue,
+            allocator=cl_tools.MemoryPool(cl_tools.ImmediateAllocator(queue)))
+
+    dim = 2
+    nel_1d = 8
+    order = 1
+
+    t_final = 3e-9
+    current_cfl = 1.0
+    velocity = np.zeros(shape=(dim,))
+    # velocity[:dim] = 1.0
+    current_dt = 1e-9
+    current_t = 0
+    constant_cfl = False
+    nstatus = 1
+    nviz = 5
+    rank = 0
+    checkpoint_t = current_t
+    current_step = 0
+    timestepper = rk4_step
+    box_ll = -0.005
+    box_ur = 0.005
+    error_state = 0
+
+    from mpi4py import MPI
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+
+    from meshmode.mesh.generation import generate_regular_rect_mesh
+    generate_grid = partial(generate_regular_rect_mesh, a=(box_ll,) * dim,
+                            b=(box_ur,) * dim, n=(nel_1d,) * dim)
+    local_mesh, global_nelements = create_parallel_grid(comm, generate_grid)
+    local_nelements = local_mesh.nelements
+
+    discr = EagerDGDiscretization(
+        actx, local_mesh, order=order, mpi_communicator=comm
+    )
+    nodes = thaw(actx, discr.nodes())
+
+    # Use Cantera for initialization
+    # -- Pick up a CTI for the thermochemistry config
+    from mirgecom.mechanisms import get_mechanism_cti
+    mech_cti = get_mechanism_cti("uiuc")
+
+    cantera_soln = cantera.Solution(phase_id="gas", source=mech_cti)
+    nspecies = cantera_soln.n_species
+    init_temperature = 1500.0
+    equiv_ratio = 1.0
+    ox_di_ratio = 0.21
+    stoich_ratio = 3.0
+    i_fu = cantera_soln.species_index("C2H4")
+    i_ox = cantera_soln.species_index("O2")
+    i_di = cantera_soln.species_index("N2")
+    x = np.zeros(nspecies)
+    x[i_fu] = (ox_di_ratio*equiv_ratio)/(stoich_ratio+ox_di_ratio*equiv_ratio)
+    x[i_ox] = stoich_ratio*x[i_fu]/equiv_ratio
+    x[i_di] = (1.0-ox_di_ratio)*x[i_ox]/ox_di_ratio
+    # one_atm = cantera.one_atm
+    one_atm = 101325.0
+
+    print(f"Input state (T,P,X) = ({init_temperature}, {one_atm}, {x}")
+    cantera_soln.TPX = init_temperature, one_atm, x
+    can_t, can_rho, can_y = cantera_soln.TDY
+    can_p = cantera_soln.P
```

Comment:
This cantera stuff is mumbo jumbo to me. Could you add some comments to explain?

Reply:
Does this help? (bcafb6e)

Reply:
Yup, thanks!